Forms/MainForm.cs

```diff
@@ -684,92 +749,29 @@ public void InitializeSettings()
         #region Randomization
 
         /// <summary>
-        /// Try to perform randomization and make rom
+        /// Try to create a RandomizedResult
         /// </summary>
-        private void TryRandomize(BackgroundWorker worker, DoWorkEventArgs e)
+        private bool TryRandomize(BackgroundWorker worker, out RandomizedResult randomized)
         {
-            if (!_settings.GenerateROM && !_settings.GenerateSpoilerLog && !_settings.GeneratePatch)
-            {
-                MessageBox.Show($"No output selected", "Error",
-                    MessageBoxButtons.OK, MessageBoxIcon.Error);
-                return;
-            }
-
-            RandomizedResult randomized;
-            if (string.IsNullOrWhiteSpace(_settings.InputPatchFilename))
-            {
-                try
-                {
-                    randomized = _randomizer.Randomize(worker, e);
-                }
-                catch (InvalidDataException ex)
-                {
-                    MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-
-                }
-                catch (Exception ex)
-                {
-                    string nl = Environment.NewLine;
-                    MessageBox.Show($"Error randomizing logic: {ex.Message}{nl}{nl}Please try a different seed", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-                }
-                
-                if (_settings.GenerateSpoilerLog
-                    && _settings.LogicMode != LogicMode.Vanilla)
-                {
-                    SpoilerUtils.CreateSpoilerLog(randomized, _settings);
-                }
-            }
-            else
+            randomized = null;
```

Comment:
Would not explicitly setting this null cause compiler error? Or is it null by default?

Reply:
It would not compile it not all code paths set an `out` parameter.